components/Flipper/FlipperFilter/FlipperFilter.tsx

```diff
@@ -81,84 +62,57 @@ function FlipperFilter(props: Props) {
         props.onChange(filter)
     }
 
-    function onOnlyBinChange(event: ChangeEvent<HTMLInputElement>) {
-        setOnlyBin(event.target.checked)
-        let filter = getCurrentFilter()
-        filter.onlyBin = event.target.checked
-        api.setFlipSetting('onlyBin', event.target.checked)
-        onFilterChange(filter)
-    }
-
-    function onOnlyUnsoldChange(event: ChangeEvent<HTMLInputElement>) {
-        let isActive = event.target.checked
-        updateOnlyUnsold(isActive)
-    }
-
-    function onMinProfitChange(value: NumberFormatValues) {
-        let val = value.floatValue || 0
-        setMinProfit(val)
-        let filter = getCurrentFilter()
-        filter.minProfit = val
-        api.setFlipSetting('minProfit', val)
-        onFilterChange(filter)
-    }
-
-    function onMinProfitPercentChange(value: NumberFormatValues) {
-        let val = value.floatValue || 0
-        setMinProfitPercent(val)
-        let filter = getCurrentFilter()
-        filter.minProfitPercent = val
-        api.setFlipSetting('minProfitPercent', val)
-        onFilterChange(filter)
-    }
-
-    function onMaxCostChange(value: NumberFormatValues) {
-        let val = value.floatValue || 0
-        setMaxCost(val)
-        let filter = getCurrentFilter()
-        filter.maxCost = val
-        api.setFlipSetting('maxCost', val)
-        onFilterChange(filter)
-    }
-
-    function updateOnlyUnsold(isActive: boolean) {
-        setOnlyUnsold(isActive)
-        let filter = getCurrentFilter()
-        filter.onlyUnsold = isActive
-        api.setFlipSetting('showHideSold', isActive)
-        onFilterChange(filter)
-    }
-
-    function onMinVolumeChange(value: NumberFormatValues) {
-        let val = value.floatValue || 0
-        setMinVolume(val)
-        let filter = getCurrentFilter()
-        filter.minVolume = val
-        api.setFlipSetting('minVolume', val)
+    function onSettingsChange(key: string, value: any, apiKey?: string) {
+        let filter = flipperFilter
+        filter[key] = value
+        api.setFlipSetting(apiKey || key, value)
+        setFlipperFilter(flipperFilter)
         onFilterChange(filter)
     }
 
     function onRestrictionsChange(restrictions: FlipRestriction[], type: 'blacklist' | 'whitelist') {
-        setRestrictions(restrictions)
         api.setFlipSetting(type, mapRestrictionsToApiFormat(restrictions.filter(restriction => restriction.type === type)))
     }
 
-    function onFreePremiumComplete() {
-        setFreePremiumFilters(true)
+    function numberFieldMaxValue(value: number = 0, maxValue: number) {
+        return value <= maxValue
     }
 
-    function onFreeLoginComplete() {
-        setFreeLoginFilters(true)
+    function isCurrentCalculationBasedOnLbin() {
+        return flipCustomizeSettings.finders?.length === 1 && flipCustomizeSettings.finders[0] === 2
```

Comment:
I am not sure why but this doesn't work, when selecting only the `sniper` in `settings` it still displays median